prow/config/config.go

```diff
@@ -2853,19 +2853,12 @@ func (repo OrgRepo) String() string {
 // NewOrgRepo creates a OrgRepo from org/repo string
 func NewOrgRepo(orgRepo string) *OrgRepo {
 	// Trim off http:// or https:// to support gerrit
-	orgRepo = trimRepoPrefix(orgRepo)
-	parts := strings.Split(orgRepo, "/")
-	switch len(parts) {
-	case 1:
-		return &OrgRepo{Org: parts[0]}
-	case 2:
-		return &OrgRepo{Org: parts[0], Repo: parts[1]}
-	//Support Gerrit repos with name Host/Repo/Name
-	case 3:
-		return &OrgRepo{Org: parts[0], Repo: strings.Join(parts[1:3], "/")}
-	default:
-		return nil
+	org, repo, err := SplitRepoName(orgRepo)
+	// Unable to split to Org/Repo
+	if err != nil {
```

Comment:
I understand that `SplitRepoName` only returns error when it can not be splitted, because I just reviewed the code yesterday. I bet I will be very confused when looking at this code some time later and ask: why do we return something when there is an error?

One alternative idea is to update `SplitRepoName` function to not return error at all, user should be able to decide by `repo == ""` instead of on error

Reply:
I agree with this, but SplitRepoName is used in quite a few places. I will make this change in a follow up PR so that this PR isn't too large

Reply:
sure, please add a comment here explaining clearly why returning valid struct when there is an error

Reply:
Done